crates/release_plz/src/config.rs

```diff
@@ -73,37 +54,88 @@ impl Config {
         }
         Ok(())
     }
+}
 
-    pub fn fill_release_config(
-        &self,
-        allow_dirty: bool,
-        no_verify: bool,
-        release_request: ReleaseRequest,
-    ) -> ReleaseRequest {
-        let mut default_config = self.workspace.packages_defaults.clone();
-        if no_verify {
-            default_config.publish_no_verify = Some(true);
-        }
-        if allow_dirty {
-            default_config.publish_allow_dirty = Some(true);
-        }
-        let mut release_request =
-            release_request.with_default_package_config(default_config.into());
+/// An object that can be configured with workspace default and package specific configs.
+pub trait Configurable {
+    /// Contains the overrides (for example, from the command-line) to apply on top of the
+    /// provided configs.
+    type Overrides;
+
+    /// Configures `self` with the given workspace default config and overrides.
+    /// Returns the new configured value.
+    fn configure_default(self, default_config: PackageConfig, overrides: &Self::Overrides) -> Self;
+
+    /// Configures `self` with the given package-specific config and overrides for a package with
+    /// the given name. Returns the new configured value.
+    fn configure_package(
+        self,
+        package: impl Into<String>,
+        package_config: PackageSpecificConfig,
+        overrides: &Self::Overrides,
+    ) -> Self;
+}
```

Comment:
what's the point of this trait if it doesn't specify any common functions?
Doesn't it make the code only more complex and less flexible? 🤔

Also, I wish this was introduced as a standalone PR as a "refactoring" and then the change of the behavior was done in a PR based on that one.